libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -104,111 +99,202 @@ - (void)dealloc {
-            }
-        }
-    }
-    [defaults setObject:loginHost forKey:kSFUserAccountOAuthLoginHost];
-    [defaults synchronize];
-    return loginHost;
+    return [SFSDKAuthPreferences sharedPreferences].loginHost;
 }
 
-#pragma mark - persistent properties
 - (NSSet *)scopes
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    NSArray *scopesArray = [defs objectForKey:kOAuthScopesKey] ?: [NSArray array];
-    return [NSSet setWithArray:scopesArray];
+    return [SFSDKAuthPreferences sharedPreferences].scopes;
 }
 
 - (void)setScopes:(NSSet *)newScopes
 {
-    NSArray *scopesArray = [newScopes allObjects];
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    [defs setObject:scopesArray forKey:kOAuthScopesKey];
-    [defs synchronize];
+    [SFSDKAuthPreferences sharedPreferences].scopes = newScopes;
 }
 
 - (NSString *)oauthCompletionUrl
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    NSString *redirectUri = [defs objectForKey:kOAuthRedirectUriKey];
-    return redirectUri;
+    return [SFSDKAuthPreferences sharedPreferences].oauthCompletionUrl;
 }
 
 - (void)setOauthCompletionUrl:(NSString *)newRedirectUri
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    [defs setObject:newRedirectUri forKey:kOAuthRedirectUriKey];
-    [defs synchronize];
+    [SFSDKAuthPreferences sharedPreferences].oauthCompletionUrl = newRedirectUri;
 }
 
 - (NSString *)oauthClientId
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    NSString *clientId = [defs objectForKey:kOAuthClientIdKey];
-    return clientId;
+    return [SFSDKAuthPreferences sharedPreferences].oauthClientId;
 }
 
 - (void)setOauthClientId:(NSString *)newClientId
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    [defs setObject:newClientId forKey:kOAuthClientIdKey];
-    [defs synchronize];
+   [SFSDKAuthPreferences sharedPreferences].oauthClientId = newClientId;
+}
+
+#pragma  mark - login & logout
+
+- (BOOL)handleAdvancedAuthenticationResponse:(NSURL *)appUrlResponse {
+    [SFSDKCoreLogger i:[self class] format:@"handleAdvancedAuthenticationResponse %@",[appUrlResponse description]];
+    
+    //should return the shared instance for advanced auth
+    SFSDKOAuthClient *client = [SFSDKOAuthClient nativeBrowserAuthInstance:nil];
+    return [client handleURLAuthenticationResponse:appUrlResponse];
+}
+
+- (BOOL)loginWithCompletion:(SFUserAccountManagerSuccessCallbackBlock)completionBlock failure:(SFUserAccountManagerFailureCallbackBlock)failureBlock {
+    SFOAuthCredentials *clientCredentials = [self retrieveClientCredentials];
+    return [self refreshCredentials:clientCredentials completion:completionBlock failure:failureBlock];
+}
+
+- (BOOL)refreshCredentials:(SFOAuthCredentials *)credentials completion:(SFUserAccountManagerSuccessCallbackBlock)completionBlock failure:(SFUserAccountManagerFailureCallbackBlock)failureBlock {
```

Comment:
Can we rename this?  Given that it's ultimately called from every login method, the naming is confusing.  For a set of new credentials, it's not going to refresh the credentials.

Also, the analogous method in `SFAuthenticationManager` explicitly guarded against uninitialized credentials in this method, basically reserving it strictly for auth refresh scenarios.